bitnami/kafka/values.yaml

```diff
@@ -342,6 +342,39 @@ auth:
     ## ref: https://docs.confluent.io/current/kafka/authentication_ssl.html#optional-settings
     ##
     endpointIdentificationAlgorithm: https
+  ## Zookeeper client configuration for kafka brokers
```

Comment:
From my point of view, I think this configuration does not belong here. This is not really something we want to directly configure on Kafka. Instead, this is more of a "reactive" change: we need to configure Kafka to use TLS because the underlying `zookeeper` is using it. As such, this should belong in the `zookeeper:` entry in the `values.yaml`.

Another good point to it is that this chart is using the zookeeper subchart by default (again, the `zookeeper:` entry in `values.yaml`). Imagine that you wanted to configure TLS for Kafka-Zookeeper communications --> You would need to do it both in `auth.zookeeper.tls` and `zookeeper.tls` with this approach.

This logic already exists in the zookeeper subchart, so we just need to move the entries under `zookeeper:` and make this chart "react" to that configuration. For example, take a look at how the authentication credentials for `redis` are configured in the `bitnami/argo-cd` chart, and how there is logic to detect which is the right `secret` to use:

https://github.com/bitnami/charts/blob/fdd8b691290c8f039d192a2b7c6f87706e7c2390/bitnami/argo-cd/values.yaml#L2325-L2333
https://github.com/bitnami/charts/blob/fdd8b691290c8f039d192a2b7c6f87706e7c2390/bitnami/argo-cd/templates/_helpers.tpl#L143-L153
https://github.com/bitnami/charts/blob/fdd8b691290c8f039d192a2b7c6f87706e7c2390/bitnami/argo-cd/templates/repo-server/deployment.yaml#L114-L121

Does it make sense to you? WDYT?

Reply:
Hi @joancafom,
thank you for the review and sorry for the late reply!  
I think I do not completely agree with this point, let me explain. 

> This is not really something we want to directly configure on Kafka. Instead, this is more of a "reactive" change: we need to configure Kafka to use TLS because the underlying zookeeper is using it. 

In my opinion, this is something we want to configure directly on Kafka. Of course, we need to configure it because zookeeper uses TLS, but we also need to configure it for `mtls` between Kafka and Zookeeper. You can configure zookeeper to accept TLS connection without configuring `mtls` in Kafka. 

> Another good point to it is that this chart is using the zookeeper subchart by default (again, the zookeeper: entry in values.yaml). Imagine that you wanted to configure TLS for Kafka-Zookeeper communications --> You would need to do it both in auth.zookeeper.tls and zookeeper.tls with this approach.

To my mind, even if we configure these parameters in the `zookeeper:` entry, you still need to configure 2 keystore/truststore. One for zookeeper, and one for kafka (mtls). You do not need/want to use the same keystore/truststore for both. So one entry will only impact Zookeeper, while the second will only impact Kafka. Of course there are some links between these two configurations because of the truststore, but it is two different configurations.

> This logic already exists in the zookeeper subchart, so we just need to move the entries under zookeeper: and make this chart "react" to that configuration. For example, take a look at how the authentication credentials for redis are configured in the bitnami/argo-cd chart, and how there is logic to detect which is the right secret to use:

If we do this, you will use the same keystore/truststore for both server and client. I think this is not a good practice. What do you think?

Reply:
Sorry for the delay. Let me discuss it with a teammate and I'll get back to you shortly, I think you're right 😁

Reply:
Thank you! Waiting for your feedback :)

Reply:
After some brief conversation, I think it does make sense to have these two entries as you originally implemented them 😁 Not only will that allow to set different keystore/trustore for Kafka & Zookeeper, but it will also allow the use of TLS with an external zookeeper instance. 

Let me recheck the PR for further comments now :) Thanks!!